### PR TITLE
Fix: Improve code coverage

### PR DESCRIPTION
**Pull request by molant**
Changes:

|           | Statements | Branch | Functions | Lines |
|-------|--------------|---------|------------|-------|
| Before| 83.6% | 73.5%   | 78.89% | 83.6% |
| Now | 85.09% | 76.08% | 81.84% | 85.09% |

### CODE REVIEW

**Comment by molant** [comment]
The main offenders are the connectors, which were tested heavily via the rules. Exception is `local` that @sarvaje is modifying in his work for #263.
Will see if there's another easy way to get more coverage but otherwise I'll change the threshold for this project.

File: packages/sonarwhal/src/lib/cli/rules/common.ts [hunk 1]
```diff
@@ -97,6 +97,10 @@ const useCases = [
 
 /** List of questions to prompt the user. */
 export const questions = (type: QuestionsType) => {
+    const noEmpty = (value) => {
```
Comment (alrra) [inline packages/sonarwhal/src/lib/cli/rules/common.ts]: Maybe use `notEmpty`?

File: packages/sonarwhal/tests/lib/utils/get-as-path-string.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import * as os from 'os';
+import * as url from 'url';
+
+import test from 'ava';
+import * as sinon from 'sinon';
+
+import { getAsPathString } from '../../../src/lib/utils/get-as-path-string';
+
+
```
Comment (alrra) [inline packages/sonarwhal/tests/lib/utils/get-as-path-string.ts]: Remove extra empty line.

File: packages/sonarwhal/tests/lib/utils/get-as-path-string.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+test.serial('getAsPathString returns the pathname of a URL if protocol is not file:', (t) => {
```
Comment (alrra) [inline packages/sonarwhal/tests/lib/utils/get-as-path-string.ts]: Maybe wrap file: in quotes? 

> `getAsPathString returns the pathname of a URL if protocol is not 'file:'`

(same goes for the other occurrences bellow)

File: packages/sonarwhal/tests/lib/utils/misc.ts [hunk 4]
```diff
@@ -27,22 +27,117 @@ const resolve = (route) => {
...
+        'my-header': '  Something  ',
+        'my-other-header': ' Another'
+    };
+
+    const myHeader = misc.getHeaderValueNormalized(headers, 'my-header2', 'missing');
+
+    t.is(myHeader, 'missing', `getHeaderValueNormalized doesn't trim and lowerCase the value`);
+});
+
+test('hasProtocol checks if a url has uses the given protocol', (t) => {
```
Comment (alrra) [inline packages/sonarwhal/tests/lib/utils/misc.ts]: For consistency with the rests, use `URL`.

File: packages/sonarwhal/tests/lib/utils/misc.ts [hunk 5]
```diff
@@ -27,22 +27,117 @@ const resolve = (route) => {
...
+    const containsProtocol = misc.hasProtocol(url, 'https:');
+    const doesnotContainProtocol = misc.hasProtocol(url, 'ftp:');
+
+    t.true(containsProtocol, `hasProtocol doesn't detect correctly the protocol https:`);
+    t.false(doesnotContainProtocol, `hasProtocol doesn't detect correctly the protocol ftp:`);
+});
+
+test('isDataUri detects if the url is a data URI or not', (t) => {
+    const noDataUri = 'https://myresource.com/';
+    const dataUri = 'data://somethinghere';
```
Comment (alrra) [inline packages/sonarwhal/tests/lib/utils/misc.ts]: Data URIs have a different [syntax](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/Data_URIs#Syntax). Maybe use a real data URI, e.g.: `data:text/plain;base64,SGVsbG8sIFdvcmxkIQ%3D%3D`?

File: packages/sonarwhal/tests/lib/utils/misc.ts [hunk 6]
```diff
@@ -27,22 +27,117 @@ const resolve = (route) => {
...
+
+    t.false(misc.isDataURI(noDataUri), `isDataUri doesn't detect correctly ${noDataUri} is not a data URI`);
+    t.true(misc.isDataURI(dataUri), `isDataUri doesn't detect correctly ${dataUri} is a data URI`);
+});
+
+test('isLocalFile detects if the url is local or not', (t) => {
```
Comment (alrra) [inline packages/sonarwhal/tests/lib/utils/misc.ts]: > URL

File: packages/sonarwhal/tests/lib/utils/misc.ts [hunk 7]
```diff
@@ -27,22 +27,117 @@ const resolve = (route) => {
...
+test('hasProtocol checks if a url has uses the given protocol', (t) => {
+    const url = 'https://myresource.com/';
+    const containsProtocol = misc.hasProtocol(url, 'https:');
+    const doesnotContainProtocol = misc.hasProtocol(url, 'ftp:');
+
+    t.true(containsProtocol, `hasProtocol doesn't detect correctly the protocol https:`);
+    t.false(doesnotContainProtocol, `hasProtocol doesn't detect correctly the protocol ftp:`);
+});
+
+test('isDataUri detects if the url is a data URI or not', (t) => {
```
Comment (alrra) [inline packages/sonarwhal/tests/lib/utils/misc.ts]: > URL

File: packages/sonarwhal/tests/lib/utils/misc.ts [hunk 8]
```diff
@@ -27,22 +27,117 @@ const resolve = (route) => {
...
+    const htmlResponse = { 'content-type': 'text/html' };
+    const noHtmlResponse = { 'content-type': 'text/javascript' };
+    const invalidContentType = { 'content-type': 'asdasdasda' };
+
+    t.true(misc.isHTMLDocument(url, htmlResponse), `isHTMLDocument doesn't recognize HTML if header is text/html`);
+    t.false(misc.isHTMLDocument(url, noHtmlResponse), `isHTMLDocument doesn't recognize is not HTML if header is text/javascript`);
+    t.false(misc.isHTMLDocument(url, invalidContentType), `isHTMLDocument doesn't recognize invalid content types are not HTML`);
+});
+
+test('isHTTP detects if the url is HTTP or not', (t) => {
```
Comment (alrra) [inline packages/sonarwhal/tests/lib/utils/misc.ts]: > URL

File: packages/sonarwhal/tests/lib/utils/misc.ts [hunk 9]
```diff
@@ -27,22 +27,117 @@ const resolve = (route) => {
...
+    const noHttpUri = 'https://myresource.com/';
+    const httpUri = 'http://somethinghere';
+
+    t.false(misc.isHTTP(noHttpUri), `isHTTP doesn't detect correctly ${noHttpUri} is not a HTTP URI`);
+    t.true(misc.isHTTP(httpUri), `isHTTP doesn't detect correctly ${httpUri} is a HTTP URI`);
+});
+
+test('isHTTPS detects if the url is HTTP or not', (t) => {
```
Comment (alrra) [inline packages/sonarwhal/tests/lib/utils/misc.ts]: > URL

**Comment by alrra** [comment]
Merged in: https://github.com/sonarwhal/sonarwhal/commit/f78b330b76382ea2420ddf848c691f1a26b13e3d.

**Pull request closed without merging**